app/lib/domicile_resolver.rb

```diff
@@ -1,27 +1,45 @@
 class DomicileResolver
-  def self.hesa_code_for_country(iso_country_code)
-    case iso_country_code
-    when nil then 'ZZ'
-    when 'AQ' then 'XX'
-    when 'CY' then 'XC'
-    when 'XK' then 'QO'
-    else
-      iso_country_code
+  class << self
+    def hesa_code_for_country(iso_country_code)
+      case iso_country_code
+      when nil then 'ZZ'
+      when 'AQ' then 'XX'
+      when 'CY' then 'XC'
+      when 'XK' then 'QO'
+      else
+        iso_country_code
+      end
     end
-  end
 
-  def self.hesa_code_for_postcode(uk_postcode)
-    prefix = uk_postcode.scan(/^[a-zA-Z]+/).first if uk_postcode.present?
+    def hesa_code_for_postcode(uk_postcode)
+      prefix = uk_postcode.scan(/^[a-zA-Z]+/).first if uk_postcode.present?
+
+      case prefix
+      when nil then 'ZZ'
+      when *POSTCODE_PREFIXES['England'] then 'XF'
+      when *POSTCODE_PREFIXES['Wales'] then 'XI'
+      when *POSTCODE_PREFIXES['Scotland'] then 'XH'
+      when *POSTCODE_PREFIXES['Northern Ireland'] then 'XG'
+      when *POSTCODE_PREFIXES['Channel Islands'] then 'XL'
+      else
+        'XK'
```

Comment:
could we add a case for this code in the country for hesa code?

Reply:
@cwrw good catch https://github.com/DFE-Digital/apply-for-teacher-training/pull/6964/commits/d06c6f811a14ad02272bb535fdf019ae6ead5f7a